src/components/scenes/FioAddressSettingsScene.js

```diff
@@ -0,0 +1,200 @@
+    return parseFloat(truncateDecimals(bns.div(val, this.props.denominationMultiplier, DIVIDE_PRECISION), 6))
+  }
+
+  onNextPress = async () => {
+    const { fioWallet, fioAddressName, isConnected, refreshAllFioAddresses } = this.props
+    const { renewalFee } = this.state
+
+    if (!isConnected) {
+      showError(s.strings.fio_network_alert_text)
+      return
+    }
+
+    if (!fioWallet) {
+      showError(s.strings.fio_wallet_missing_for_fio_address)
+      return
+    }
+
+    if (renewalFee === null) {
+      showError(s.strings.fio_get_fee_err_msg)
+      return
+    }
+
+    try {
+      this.toggleRenewLoading(true)
+      const { expiration } = await renewFioAddress(fioWallet, fioAddressName, renewalFee)
+
+      refreshAllFioAddresses()
+
+      showToast(s.strings.fio_request_renew_ok_text)
+      Actions.pop()
+      window.requestAnimationFrame(() => {
+        Actions.refresh({ fioAddressName, expiration: intl.formatExpDate(expiration) })
+      })
+    } catch (e) {
+      showError(e.message)
+    }
+    this.toggleRenewLoading()
+  }
+
+  renderFeeAndBalance() {
+    const { feeLoading, displayFee, balance, renewalFee } = this.state
+
+    if (feeLoading) return <ActivityIndicator style={styles.activityIndicator} size="small" />
+
+    if (renewalFee === null && !feeLoading) {
+      return (
+        <View>
+          <T style={styles.title}>{s.strings.fio_get_fee_err_msg}</T>
+        </View>
+      )
+    }
+
+    return (
+      <View style={styles.texts}>
+        <View style={styles.spacer} />
+        <View style={styles.spacer} />
+        <T style={styles.title}>{s.strings.fio_renew_fee_label}</T>
+        <T style={styles.content}>
+          {displayFee ? `${displayFee} ${s.strings.fio_address_confirm_screen_fio_label}` : s.strings.fio_address_confirm_screen_free_label}
+        </T>
+        <View style={styles.spacer} />
+        {displayFee ? (
+          <View>
+            <View style={styles.balanceText} />
+            <T style={styles.title}>{s.strings.fio_address_confirm_screen_balance_label}</T>
+            <T style={displayFee > balance ? styles.balanceTitleDisabled : styles.balanceTitle}>
+              {balance ? balance.toFixed(2) : '0'} {balance ? s.strings.fio_address_confirm_screen_fio_label : ''}
+            </T>
+          </View>
+        ) : null}
+      </View>
+    )
+  }
+
+  render() {
+    const { fioAddressName, expiration } = this.props
+    const { feeLoading, displayFee, balance, renewalFee, renewLoading } = this.state
+
+    return (
+      <SceneWrapper>
+        <View style={styles.info}>
+          <T style={styles.title}>{s.strings.fio_address_register_form_field_label}</T>
+          <T style={styles.content}>{fioAddressName}</T>
+        </View>
+        <View style={styles.info}>
+          <T style={styles.title}>{s.strings.fio_address_details_screen_expires}</T>
+          <T style={styles.content}>{expiration}</T>
+        </View>
+        {this.renderFeeAndBalance()}
```

Comment:
show the 'Renew Address/Domain' button, and expand slider with confirmation when click on it

Reply:
done